content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,367 @@
+
+TODO: Add picture of nodes quorum finding
+
+Nodes know about other nodes' quorum slices in the same manner they know about rest of the information about their network peers — from the broadcast information that every node sends to the network whenever its voting state changes.
+Each broadcast message includes the details of the sending node's IP and its quorum slices nodes - a typical implementation used in FBA systems (and generally in other blockchains) is [Gossip protocol](https://en.wikipedia.org/wiki/Gossip_protocol).
+
+FBA systems create confidence in the outcome of a decentralized vote by running a process called `federated voting`.
+Federated voting is a procedure for discovering whether a network of participants can agree on a proposal.
+In the cryptocurrency world a proposal is a ledger state with included transactions.
+In a round of a federated voting process, each node must choose one of potentially many possible values as the outcome of that round.
+It cannot do so until it ensures that the other nodes in the network will not choose any different outcome.
+To be consistent with the federated voting algorithm, the nodes exchange a flurry of messages allowing each of them to confirm that a quorum of nodes accepts the same vote.
+
+In FBA systems there is no majority existing as a number — because no one knows the total size of the network, but the concept of majority is not even needed.
+Since the system is open, it is possible someone to try to gain influence of the network by conducting a so-called Sybil attack.
+A [Sybil attack](https://www.sciencedirect.com/topics/computer-science/sybil-attack)) is an attempt to control a peer network by creating multiple fake identities.
+To outside observers, these fake identities appear to be unique peers.
+As a result that entity can influence the network through additional voting power in a democratic network.
+
+In order for the federated voting to work, one possible solution (implemented by Stellar) is to require a network property called "quorum intersection".
+This means that any two quorums that can be constructed in the network with this property, will always share at least one node.
+For determining the prevailing sentiment of the network, this is as good as having a majority.
+Intuitively, it means that if any quorum agrees to statement `X`, no other quorum can ever agree to `not-X`, because it will necessarily include some node from the first quorum that has already voted for X.
+This is the transitive closure property that this type of systems typically use to fight against Sybil attacks.
+
+The overlapping nodes could be all Byzantine nodes.
+In that case, having quorum intersection does not help the network to agree at all.
+For that reason, the most important assumption about quorum intersection model is that the network enjoys quorum intersection `afer` all misbehaving nodes are removed from the network.
```

Comment:
Why does it say `afer`, and not just _after_?

Reply:
Because I missed the 't'

Reply:
This has been rephrased to simply this https://github.com/optakt/optakt.github.io/pull/21/commits/ca80e6c74bee64b8ce712ca515bb205ee645d971#diff-4bf8323c6426a93da1ce22be1b8528779ef6b61fdacef3e0d065f1e1dd47bea5R235